docs/data/toolpad/studio/how-to-guides/railway-deploy.md

```diff
@@ -0,0 +1,52 @@
+# Deploy to Railway
+
+<p class="description">You can host and share your Toolpad Studio apps on Railway in a few minutes, for free.</p>
+
+This guide walks you throught the deployment of [qr-generator](https://github.com/mui/mui-toolpad/tree/master/examples/qr-generator) example from the Toolpad repository.
+
+## Prerequisites
+
+- A [Railway](https://railway.app/) account
+- A [GitHub](https://github.com) account
+- A GitHub repository containing Toolpad Studio app. Check out [pushing Toolpad Studio app to GitHub](/toolpad/studio/how-to-guides/render-deploy/#pushing-your-toolpad-studio-app-to-github) for this step.
+
+## Creating a new app on Railway
+
+1. From your Railway dashboard, click on **New Project** from the top-right corner and choose **Deploy from GitHub repo** form the drop-down.
+
+{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-railway/railway-create-new-project.png", "alt": "Railway new project", "caption": "Creating a new Railway project", "indent": 1 }}
+
+2. Depending on whether the visibility of the repository is set to private or public, you may need to connect your GitHub account to Railway using **Configure GitHub App**. Once done, you'll see a list of repositories to choose from.
+
+{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-railway/choose-repository.png", "alt": "Choosing the repository", "caption": "Choosing the repository", "indent": 1 }}
+
+3. Select the repository and click **Deploy Now**. Railway can guess that you are deploying a Node app.
+
+4. The default build may fail as some config changes are required. On the deployment screen, you'll see four tabs, navigate to the **Settings** tab and update the root directory as shown below.
```

Comment:
Can we create an example where the default deployment does not fail, and talk about this as a special case at the end? I think this creates an expectation that you need to always change the default root directory when deploying a Toolpad app, which is not correct.

I'd recommend creating a new repository with a Toolpad app and using that to create the deployment guide. Creating a FAQs or Advanced section at the end where you can discuss what to do if your app fails, which can include variable setting and root directory configuration.

Reply:
> I think this creates an expectation that you need to always change the default root directory when deploying a Toolpad app, which is not correct.

The default deployment also fails due to the PORT variable. Regarding changing the root directory, I think it would be required in most cases, as the user may have a folder containing the toolpad app. I'll remove the 'may fail' part, but I think we should add this step to the guide. 

> I'd recommend creating a new repository with a Toolpad app. 
Yes, we can do it. I thought keeping it from examples would be ok. We can decide, and the same example folder/repository could be used for other guides.

Reply:
> > I think this creates an expectation that you need to always change the default root directory when deploying a Toolpad app, which is not correct.
> 
> The default deployment also fails due to the PORT variable. Regarding changing the root directory, I think it would be required in most cases, as the user may have a folder containing the toolpad app. I'll remove the 'may fail' part, but I think we should add this step to the guide.
> 

The `toolpad` folder doesn't have to be the root directory - in this scenario as well, you've set the root to `examples/qr-generator` and not `examples/qr-generator/toolpad`. I think we should address the happy path in the instructions and then address edge cases at the end. 

Regards the `PORT` variable, as far as I can see, Railway lets you add variables before the first deployment, and nudges you to redeploy once the variable is added:

![Screenshot 2024-04-11 at 2 33 17 PM](https://github.com/mui/mui-toolpad/assets/19550456/97126b72-8ac5-44af-a34c-ff93680ff40e)

I think we can follow this process with an independent repository, and then address as the variables and root configuration settings as FAQs in the end, something like:


### Common reasons for failed deployment

- A `PORT` variable is not set. Click the **Settings*** button and go to the **Variables** tab to add a `PORT` variable. If you haven't changed the default port on which your Toolpad Studio app runs, set the value of this variable to 3000.

- Your `root` directory is not configured. In some cases, your Toolpad Studio app might be placed inside a subdirectory of a larger repository on GitHub. In that case, navigate to the **Settings** tab and update the root directory as follows:

Reply:
An independent repository that I can use: https://github.com/Prakhar-org/qr-generator. Does this look ok? It is under an org name Prakhar-org and we'll use it for all future deployment guides.

With the above repo and setting port variable before deployment should make it a happy flow. I'll make these changes to the doc.

Reply:
An independent repository that I can use: https://github.com/Prakhar-org/qr-generator. It is under an org name Prakhar-org and we'll use it for all future deployment guides. Does this look ok? 

With the above repo, setting port variable before deployment should make it a happy flow. I'll make these changes to the doc.